examples/transaction.rs

```diff
@@ -21,23 +19,13 @@ async fn get(client: &Client, key: Key) -> Option<Value> {
     txn.get(key).await.expect("Could not get value")
 }
 
-// Ignore a spurious warning from rustc (https://github.com/rust-lang/rust/issues/60566).
-#[allow(unused_mut)]
-async fn scan(client: &Client, range: impl RangeBounds<Key>, mut limit: usize) {
+async fn scan(client: &Client, range: impl Into<BoundRange>, mut _limit: usize) {
```

Comment:
Why cut out the limit functionality here?

Reply:
The `limit` parameter has implemented at 963e3b3.

Here we've got some questions:

1. What's the maximum limit of a transactional scan? Should it be checked as what raw client does?

https://github.com/tikv/client-rust/blob/65602f7a45c9807a4119fc28488519a4ac9b8e4c/src/raw/client.rs#L249-L252

2. How should the caller pass `key_only` to client? Seems that `raw` client has a field to indicate this matter.

https://github.com/tikv/client-rust/blob/65602f7a45c9807a4119fc28488519a4ac9b8e4c/src/raw/client.rs#L15-L19

Reply:
1: Yes.

2: yes, transactional scan also has that field: https://github.com/pingcap/kvproto/blob/master/proto/kvrpcpb.proto#L50

Reply:
1. If I understand it correctly, just declare a constant `const MAX_TXN_KV_SCAN_LIMIT: u32 = 10240;` under `src/transaction/transaction.rs`and check the limit.
2. Yes, I know there is a field on Protobuf definition, but I am not quite sure how to implement it within transactional client. Should I add a field `key_only` on txn client, or `Transaction` itself?